tests/faucet_mininet_test_unit.py

```diff
@@ -3297,3 +3297,509 @@ def test_switching(self):
+        # pylint: disable=no-member
+        self.net.controller.cmd('echo {} > {}/contr_pid'.format(last_pid, self.tmpdir))
+        self.pids['faucet'] = last_pid
+
+        # pylint: disable=no-member
+        contr_num = int(self.net.controller.name.split('-')[1]) % 255
+        self.assertLess(int(contr_num), 255)
+        self.net.addLink(
+            portal,
+            self.net.controller,
+            params1={'ip': '192.168.%s.2/24' % contr_num},
+            params2={'ip': '192.168.%s.3/24' % contr_num})
+
+        portal.cmd('ping -c5 192.168.%s.3' % contr_num)
+        self.run_controller(self.net.controller)
+
+        interweb.cmd('echo "This is a text file on a webserver" > index.txt')
+        self.ws_port, _ = faucet_mininet_test_util.find_free_port(
+            self.ports_sock, self._test_name())
+
+        interweb.cmd('python -m SimpleHTTPServer {0} &'.format(self.ws_port))
+
+        self.clients = self.net.hosts[2:]
+        self.setup_hosts(self.clients)
+
+        self.start_dhcp_server(interweb, gw='10.0.0.2', dns='8.8.8.8')
+
+        self.run_hostapd(portal)
+
+
+class FaucetSingleAuthenticationMultiUserLogOnTest(FaucetAuthenticationSingleSwitchTest):
+    """Check if authenticated and unauthenticated users can communicate and of different authentication methods (1x & cp)"""
+
+    def ping_between_hosts(self, users):
+        """Ping between the specified host
+        Args:
+            users (list<mininet.host>): users to ping between.
+                0 & 1 should be authenitcated.
+                2 should be unauthenticated,
+        """
+        for user in users:
+            user.defaultIntf().updateIP()
+
+        h0 = users[0]
+        h1 = users[1]
+        h2 = users[2]
+        h1_ip = ipaddress.ip_address(unicode(h1.IP()))
+        # h2 will not have an ip via dhcp as they are unauthenticated, so give them one.
+        h2.setIP('10.0.12.253')
+        h2_ip = ipaddress.ip_address(unicode(h2.IP()))
+        # ping between the authenticated hosts
+        self.one_ipv4_ping(h0, h1_ip)
+        self.one_ipv4_ping(h1, '10.0.0.2')
+
+        #ping between an authenticated host and an unauthenticated host
+        try:
+            self.one_ipv4_ping(h1, h2_ip)
+        except AssertionError:
+            pass
+        else:
+            self.fail('{0} {1} should not be able to ping {2} {3}'.format(h1, h1.IP(), h2, h2.IP()))
+ 
+        ploss = self.net.ping(hosts=[users[0], users[2]], timeout='5')
+        self.assertAlmostEqual(ploss, 100)
+        try:
+            self.one_ipv4_ping(h0, h2_ip)
+        except AssertionError:
+            pass
+        else:
+            self.fail('{0} {1} should not be able to ping {2} {3}'.format(h0, h0.IP(), h2, h2.IP()))
+
+    def test_onlydot1x(self):
+        """Only authenticate through dot1x"""
+        users = self.clients
+
+        self.logon_dot1x(users[0])
+        self.logon_dot1x(users[1])
+        time.sleep(5)
```

Comment:
Can we avoid this sleep?
If h1 is authed the port_acl rules will be installed. so check for them?